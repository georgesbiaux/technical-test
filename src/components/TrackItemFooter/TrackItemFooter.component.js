import React from 'react';
import { FormattedMessage } from 'react-intl';
import StyledTrackItemFooter from './TrackItemFooter.style';

const TrackItemFooter = props => (
  <StyledTrackItemFooter className="track-footer">
    <div className="added-by">
      <FormattedMessage
        id="trackItem.addedBy"
        values={{
          username: props.track.adder.name,
        }}
      />
    </div>
    <div className="time-before-play">
      <FormattedMessage
        id="trackItem.in"
        values={{
          duration: Math.ceil(props.timeBeforePlay / 60),
        }}
      />
    </div>
  </StyledTrackItemFooter>
);

export default TrackItemFooter;
