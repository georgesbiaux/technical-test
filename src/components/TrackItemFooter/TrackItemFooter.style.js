import styled from 'styled-components';

const StyledTrackItem = styled.div`
  line-height: 20px;
  font-size: 12px;
  display: flex;
  flex-direction: row;

  .added-by {
    text-align: left;
    color: #74a7d2;
    flex: 1;
  }

  .time-before-play {
    padding-right: 10px;
    color: #8f8f8f;
  }
`;

export default StyledTrackItem;
