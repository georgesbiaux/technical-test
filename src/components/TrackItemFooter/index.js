export { default } from './TrackItemFooter.component';
