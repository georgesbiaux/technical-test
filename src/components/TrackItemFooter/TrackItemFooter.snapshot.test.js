import React from 'react';
import createComponentWithIntl from 'services/i18n/create-component-with-intl';
import toJSON from 'enzyme-to-json';
import TrackItemFooter from './TrackItemFooter.component';

describe('<TrackItemFooter>', () => {
  it('should render correctly', () => {
    const props = {
      track: {
        id: 1,
        name: 'Beast Of Burden',
        duration: 211,
        priority: false,
        artist: 'The Rolling Stones',
        adder: {
          id: 1,
          name: 'kant',
          pictureUrl:
            'https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd',
        },
        pictureUrl:
          'https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372',
        votes: {
          count: 0,
          userVoted: false,
        },
      },
    };
    const tree = createComponentWithIntl(<TrackItemFooter {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
