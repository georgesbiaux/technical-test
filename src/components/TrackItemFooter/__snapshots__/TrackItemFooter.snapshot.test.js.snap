// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItemFooter> should render correctly 1`] = `undefined`;
